### ISSUE TITLE
Question about reproducing the PerAct2 baseline used in a recent paper

### ISSUE THREAD

**Issue by HangyuDemo** [markusgrotz/peract_bimanual#29]
Hi, thanks for releasing PerAct2 and for making the code and checkpoints available.

I recently noticed that the paper “Gripper Keypose and Object Pointflow as Interfaces for Bimanual Robotic Manipulation” uses PerAct2 as a baseline in their experimental comparisons. I have been trying to reproduce a comparable PerAct2 baseline using the official code provided in this repository.

However, when training PerAct2 with 256×256 image inputs, I have not been able to reach the performance shown in the paper. Additionally, when evaluating using the checkpoint provided in this repo (https://drive.proton.me/urls/4AKGSX85GW#OuS59HBummxV), the performance on our side is also noticeably lower than expected.

<img width="1554" height="322" alt="Image" src="https://github.com/user-attachments/assets/1876c4a5-6cd0-4de5-9633-2f40f93820ec" />

I was wondering if you might have any guidance on things to pay special attention to when training or evaluating PerAct2 (e.g., resolution-dependent settings, data preprocessing, training length, or evaluation details). I also wanted to ask whether there is a trained PerAct2 checkpoint that more closely matches the baseline used in that paper, if sharing such a model is possible.

Any advice or pointers would be greatly appreciated. Thanks again for sharing this great work.

**Comment by markusgrotz** [comment]
Hi Hang, thank you very much for your interest and for checking this.

Those results come from the setup used in the [Gripper Keypose and Object Pointflow paper](https://github.com/InternRobotics/PPI), right? They almost certainly trained longer, may have used slightly different training details. The method itself is also stronger. I am not sure whether the authors released their checkpoints.

If you are mainly looking for strong checkpoints, I recommend taking a look at the [3D-FlowMatch-Actor work](https://3d-flowmatch-actor.github.io/). They report very impressive results and, as far as I know, have released checkpoints.

Is this helpful for you?

**Comment by HangyuDemo** [comment]
Hi Markus, thanks a lot for the detailed reply — this is very helpful.

Yes, the results I’m referring to are indeed from the Gripper Keypose and Object Pointflow paper. According to their description, they trained PerAct2 for around 100k iterations and used 256×256 image resolution, but it seems the PerAct2 code itself was largely unchanged.

In our case, we are currently training with a single GPU, so we had to set replay.batch_size = 3, which might be one of the reasons for the performance gap. As a next step, I plan to try increasing the batch size to 8 by using multiple GPUs.

However, I’ve run into an issue when enabling multi-GPU training (setting ddp.num_devices > 1): during replay buffer creation, file conflicts / synchronization issues occur, which causes the training to crash. I was wondering if you have any recommendations on how to properly handle replay buffer construction when using DDP (e.g., rank-0-only buffer creation, separate buffers per process, or other best practices).

Thanks again for your help and for maintaining the codebase — really appreciate it.

**Comment by markusgrotz** [comment]
Hi Hang, Are you using SLURM or how does your multi-GPU setup look like? The crash during the replay buffer creation seems like you are running multiple jobs in parallel that access the same directory.

**Comment by HangyuDemo** [comment]
Hi Markus, thanks for the reply!

Yes, that was indeed the issue. We were running multi-GPU jobs that accessed the same replay buffer directory in parallel. I later modified the code so that each process creates and uses its own replay buffer directory, and that resolved the crash during replay buffer creation.

However, after fixing this, we noticed that the training loss still looks quite strange (as shown in the screenshot above). For this run, we are using 2 GPUs, with replay.batch_size = 3 epoches=100000. The loss fluctuates significantly near the end of training, even though the run completes without crashing.

<img width="1052" height="608" alt="Image" src="https://github.com/user-attachments/assets/38533617-abc4-4342-9993-d17f14a0481d" />

Do you have any suggestions on what might cause this behavior? For example, could this be related to the relatively small batch size per GPU, any DDP-specific details we should double-check, or potential side effects of having each process use its own replay buffer?

**Comment by markusgrotz** [comment]
Hi Hang, Yes I suspect there is a minor configuration issue in your SLURM configuration. I need more information as I am only guessing. I think it would be easiest to walk through this together on a short call. Would that work for you? If so could you send me an email so we can set up a brief Zoom meeting?

**Comment by HangyuDemo** [comment]
Hi Markus,

I’d be happy to continue the discussion over email. My email address is yh974125@gmail.com
. I’ve already sent you an email (grotz@cs.washington.edu), and I’m looking forward to your reply!
